spec/core/middleware/request/retry_spec.rb

```diff
@@ -116,4 +116,76 @@ def runtime
       end
     end
   end
+
+  context "with instrumentation on retry" do
+    let(:instrumenter) { TestInstrumenter.new }
+
+    before do
+      client.config.instrumentation = instrumenter
+      stub_request(:get, %r{instrumented}).to_return(status: 429, headers: {retry_after: 1}).to_return(status: 200)
+    end
+
+    it "instruments retry attempts with correct payload" do
+      client.connection.get("instrumented")
+
+      retry_events = instrumenter.find_events("zendesk.retry")
+      expect(retry_events.size).to eq(1)
+
+      event = retry_events.first[:payload]
+      expect(event[:attempt]).to eq(1)
+      expect(event[:endpoint]).to eq("/api/v2/instrumented")
+      expect(event[:method]).to eq(:get)
+      expect(event[:reason]).to eq("rate_limited")
+      expect(event[:delay]).to be >= 0
+    end
+
+    xit "instruments consecutive retry attempts" do
```

Comment:
Should this and the following test be skipped?

Reply:
I meant to leave a comment. The current behavior is that "retries" happen only once. If we want to change it, I think it should be in a follow up PR. I changed the implementation from `while` to `if` to avoid an infinite loop and to keep the behavior the same.